### PR TITLE
Merge/release 3.6

### PR DESCRIPTION
**Pull request by bdmogal**
Lots of conflicts, especially in `DefaultStore`.
Running build - http://builds.cask.co/browse/CDAP-DUT4815-1

### CODE REVIEW

**Comment by bdmogal** [comment]
Green!

**Comment by gokulavasan** [comment]
LGTM 👍 @maochf Can you take a look at the DefaultStore changes to make sure it looks good?

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/DefaultStore.java [hunk 1]
```diff
@@ -987,9 +970,9 @@ private ApplicationSpecificationWithChangedFlows(ApplicationSpecification delega
   }
 
   private static ApplicationSpecification replaceWorkerInAppSpec(final ApplicationSpecification appSpec,
-                                                                 final Id.Program id,
+                                                                 final ProgramId id,
                                                                  final WorkerSpecification workerSpecification) {
-    return new ApplicationSpecificationWithChangedWorkers(appSpec, id.getId(), workerSpecification);
+    return new ApplicationSpecificationWithChangedWorkers(appSpec, id.getEntityName(), workerSpecification);
```
Comment (ecfm) [inline cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/DefaultStore.java]: What is this `getEntityName` method?

**Review by ecfm — COMMENTED** [review]
What is `getEntityName` method?

File: cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/DefaultStore.java (hunk 1 above)
Comment (bdmogal) [reply cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/DefaultStore.java]: its the same as `getProgram` for `ProgramId`, `getApplication` for `ApplicationId`, etc

Comment (ecfm) [reply cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/DefaultStore.java]: oh, ok, didn't find it on my local file. lgtm

Comment (bdmogal) [reply cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/store/DefaultStore.java]: yes, it was added on develop.

**Pull request merged**